--- /home/tanuj4/workspace/gdal-trunkcopy/gdal/frmts/png/libpng/pngrutil.c	2016-04-09 17:04:05.972129715 +0530
+++ pngrutil.c	2016-04-08 16:15:07.466714000 +0530
@@ -3013,7 +3013,7 @@
 
    /* Check for unhandled critical chunks */
    if (handled == 0 && PNG_CHUNK_CRITICAL(png_ptr->chunk_name))
-      png_chunk_error(png_ptr, "unhandled critical chunk");
+      png_chunk_warning(png_ptr, "unhandled critical chunk");
 }
 
 /* This function is called to verify that a chunk name is valid.
@@ -3040,7 +3040,7 @@
       int c = chunk_name & 0xff;
 
       if (c < 65 || c > 122 || (c > 90 && c < 97))
-         png_chunk_error(png_ptr, "invalid chunk type");
+         png_chunk_warning(png_ptr, "invalid chunk type");
 
       chunk_name >>= 8;
    }
