--- /home/even/libpng-1.2.46/pngrutil.c	2011-07-09 12:30:23.000000000 +0200
+++ pngrutil.c	2011-07-29 19:52:18.831921846 +0200
@@ -2404,7 +2404,8 @@
 #endif
         )
 #endif
-          png_chunk_error(png_ptr, "unknown critical chunk");
+          /* GDAL change : png_chunk_error -> png_chunk_warning */
+          png_chunk_warning(png_ptr, "unknown critical chunk");
    }
 
 #ifdef PNG_READ_UNKNOWN_CHUNKS_SUPPORTED
@@ -2488,7 +2489,8 @@
    if (isnonalpha(chunk_name[0]) || isnonalpha(chunk_name[1]) ||
        isnonalpha(chunk_name[2]) || isnonalpha(chunk_name[3]))
    {
-      png_chunk_error(png_ptr, "invalid chunk type");
+      /* GDAL change : png_chunk_error -> png_chunk_warning */
+      png_chunk_warning(png_ptr, "invalid chunk type");
    }
 }
 
