--- pngrutil.c.old	2020-02-23 16:47:37.962028934 +0100
+++ pngrutil.c	2020-02-23 16:51:33.653908065 +0100
@@ -3115,7 +3115,8 @@
 
    /* Check for unhandled critical chunks */
    if (handled == 0 && PNG_CHUNK_CRITICAL(png_ptr->chunk_name))
-      png_chunk_error(png_ptr, "unhandled critical chunk");
+       /* GDAL change : png_chunk_error -> png_chunk_warning */
+      png_chunk_warning(png_ptr, "unhandled critical chunk");
 }
 
 /* This function is called to verify that a chunk name is valid.
@@ -3143,7 +3144,8 @@
       int c = cn & 0xff;
 
       if (c < 65 || c > 122 || (c > 90 && c < 97))
-         png_chunk_error(png_ptr, "invalid chunk type");
+        /* GDAL change : png_chunk_error -> png_chunk_warning */
+         png_chunk_warning(png_ptr, "invalid chunk type");
 
       cn >>= 8;
    }
